metrics/factory.go

```diff
@@ -17,16 +17,19 @@ func NewMetricsRunnerFromDSN(DSN string) (Metrics, contextx.Runner) {
 	}
 
 	params := URL.Query()
-	namespace := params.Get("namespace")
-	if URL.Scheme == "datadog" && namespace == "" {
-		panic("datadog metrics need a namespace")
-	}
 
 	// publisher is both Metrics and Runner
 	var publisher *Publisher
+	namespace := params.Get("namespace")
 	switch URL.Scheme {
 	case "datadog":
-		publisher = NewDataDog()
+		if namespace == "" {
+			panic("datadog metrics need a namespace")
+		}
+		publisher = NewDataDog(
+			WithDDHost(params.Get("host")),
```

Comment:
If host is not defined, it will set  host = "", no?  I think you should check if the parameter exists and, if it exists, set it. 
Same for the port.

Reply:
https://github.com/socialpoint-labs/bsk/pull/49/files#diff-29b845f0c4b074220389b6c831360468R102